src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -61,15 +60,8 @@ public Queue(IEnumerable<T> collection)
             if (collection == null)
                 throw new ArgumentNullException("collection");
 
-            _array = new T[DefaultCapacity];
-
-            using (IEnumerator<T> en = collection.GetEnumerator())
-            {
-                while (en.MoveNext())
-                {
-                    Enqueue(en.Current);
-                }
-            }
+            _array = EnumerableHelpers.ToArray(collection, out _size);
```

Comment:
@ellismg, any concerns with this?  As Justin mentions, I'd previously avoided doing this, but we've made other changes in corefx along these lines (i.e. adding interface specialization where it wasn't before), and I'm ok with it.